ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/ForeignKeyTableTableJoinBuilder.java

```diff
@@ -29,28 +34,45 @@ private ForeignKeyTableTableJoinBuilder() {
   public static <KLeftT, KRightT> KTableHolder<KLeftT> build(
       final KTableHolder<KLeftT> left,
       final KTableHolder<KRightT> right,
-      final ForeignKeyTableTableJoin<KLeftT, KRightT> join
+      final ForeignKeyTableTableJoin<KLeftT, KRightT> join,
+      final RuntimeBuildContext buildContext
   ) {
     final LogicalSchema leftSchema = left.getSchema();
     final LogicalSchema rightSchema = right.getSchema();
 
     final ForeignKeyJoinParams<KRightT> joinParams = ForeignKeyJoinParamsFactory
         .create(join.getLeftJoinColumnName(), leftSchema, rightSchema);
 
+    final Formats leftFormats = join.getLeftFormats();
+
+    final PhysicalSchema physicalSchema = PhysicalSchema.from(
+        joinParams.getSchema(),
+        leftFormats.getKeyFeatures(),
+        leftFormats.getValueFeatures()
+    );
+
+    final Serde<GenericRow> valSerde = buildContext.buildValueSerde(
+        leftFormats.getValueFormat(),
+        physicalSchema,
+        join.getProperties().getQueryContext()
+    );
+
     final KTable<KLeftT, GenericRow> result;
     switch (join.getJoinType()) {
-      case LEFT:
-        result = left.getTable().leftJoin(
+      case INNER:
+        result = left.getTable().join(
             right.getTable(),
             joinParams.getKeyExtractor(),
-            joinParams.getJoiner()
+            joinParams.getJoiner(),
+            Materialized.with(null, valSerde)
```

Comment:
Just checking: is it correct to leave the key serde as null and only pass in value serde? Does this reset any key serde information that is already present?